web/packages/extension/src/ruffle.ts

```diff
@@ -57,6 +58,60 @@ function openInNewTab(swf: URL): void {
     window.postMessage(message, "*");
 }
 
+function polyfillCurrentScript(): void {
+    const script = document.currentScript!;
```

Comment:
Instead of using a complicated new function below (`determineKnownTypeHasNoEmptyStringProp`), couldn't we just replace this with this code?

```js
    const script = document.currentScript as HTMLScriptElement | null;
    if (!script?.src) return;
```

Reply:
Edited according to your other comment:

```js
    const script = document.currentScript;
    if (!(script instanceof HTMLScriptElement) || !script.src) return;
```